### ISSUE TITLE
Adding new Schema Extractor to existing Schema is failing with a HTTP 500 error

### ISSUE THREAD

**Issue by johnaohara** [Hyperfoil/Horreum#71]
Trying to add a new Schema Extractor to and existing schema, that is connected to runs containing existing data, is failing with the following errors;

```
2021-08-11 11:53:08,942 WARN  [com.arj.ats.arjuna] (Transaction Reaper) ARJUNA012117: TransactionReaper::check processing TX 0:ffff0a580045:96b7:6111207e:285b in state  RUN
2021-08-11 11:53:08,944 WARN  [com.arj.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012095: Abort of action id 0:ffff0a580045:96b7:6111207e:285b invoked while multiple threads active within it.
2021-08-11 11:53:08,944 WARN  [com.arj.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012381: Action id 0:ffff0a580045:96b7:6111207e:285b completed with multiple threads - thread executor-thread-140 was in progress with java.base@11.0.5/java.net.SocketInputStream.socketRead0(Native Method)
java.base@11.0.5/java.net.SocketInputStream.socketRead(SocketInputStream.java:115)
java.base@11.0.5/java.net.SocketInputStream.read(SocketInputStream.java:168)
java.base@11.0.5/java.net.SocketInputStream.read(SocketInputStream.java:140)
org.postgresql.core.VisibleBufferedInputStream.readMore(VisibleBufferedInputStream.java:161)
org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:128)
org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:113)
org.postgresql.core.VisibleBufferedInputStream.read(VisibleBufferedInputStream.java:73)
org.postgresql.core.PGStream.receiveChar(PGStream.java:443)
org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2056)
org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:322)
org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:130)
io.agroal.pool.wrapper.PreparedStatementWrapper.executeUpdate(PreparedStatementWrapper.java:88)
org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3574)
org.hibernate.persister.entity.AbstractEntityPersister.updateOrInsert(AbstractEntityPersister.java:3436)
org.hibernate.persister.entity.AbstractEntityPersister.update(AbstractEntityPersister.java:3850)
org.hibernate.action.internal.EntityUpdateAction.execute(EntityUpdateAction.java:201)
org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:604)
org.hibernate.engine.spi.ActionQueue.lambda$executeActions$1(ActionQueue.java:478)
org.hibernate.engine.spi.ActionQueue$$Lambda$930/0x0000000800ac4440.accept(Unknown Source)
java.base@11.0.5/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:475)
org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:344)
org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:40)
org.hibernate.internal.SessionImpl$$Lambda$747/0x00000008009b1040.accept(Unknown Source)
org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107)
org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1391)
org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1378)
io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.flush(TransactionScopedSession.java:213)
io.quarkus.hibernate.orm.runtime.session.ForwardingSession.flush(ForwardingSession.java:93)
org.hibernate.Session_5b93bee577ae2f8d76647de04cfab36afbf52958_Synthetic_ClientProxy.flush(Session_5b93bee577ae2f8d76647de04cfab36afbf52958_Synthetic_ClientProxy.zig:2761)
io.hyperfoil.tools.horreum.svc.SchemaServiceImpl.add(SchemaServiceImpl.java:113)
io.hyperfoil.tools.horreum.svc.SchemaServiceImpl_Subclass.add$$superforward1(SchemaServiceImpl_Subclass.zig:512)
io.hyperfoil.tools.horreum.svc.SchemaServiceImpl_Subclass$$function$$2.apply(SchemaServiceImpl_Subclass$$function$$2.zig:33)
io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
io.quarkus.security.runtime.interceptor.SecurityHandler.handle(SecurityHandler.java:24)
io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor.intercept(RolesAllowedInterceptor.java:29)
io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor_Bean.intercept(RolesAllowedInterceptor_Bean.zig:386)
io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:50)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInCallerTx(TransactionalInterceptorBase.java:302)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:40)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:57)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:32)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:340)
io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
io.hyperfoil.tools.horreum.svc.SchemaServiceImpl_Subclass.add(SchemaServiceImpl_Subclass.zig:808)
java.base@11.0.5/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
java.base@11.0.5/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
java.base@11.0.5/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.base@11.0.5/java.lang.reflect.Method.invoke(Method.java:566)
org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
org.jboss.resteasy.core.ResourceMethodInvoker$$Lambda$912/0x0000000800a93c40.get(Unknown Source)
org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:408)
org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:69)
org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
org.jboss.resteasy.core.SynchronousDispatcher$$Lambda$900/0x0000000800a65840.run(Unknown Source)
org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
org.jboss.resteasy.core.SynchronousDispatcher$$Lambda$901/0x0000000800a65c40.get(Unknown Source)
org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:249)
io.quarkus.resteasy.runtime.ResteasyFilter.doFilter(ResteasyFilter.java:35)
io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
io.hyperfoil.tools.horreum.server.RouteFilter.doFilter(RouteFilter.java:28)
io.hyperfoil.tools.horreum.server.RouteFilter_ClientProxy.doFilter(RouteFilter_ClientProxy.zig:284)
io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
io.hyperfoil.tools.horreum.server.UserTeamsFilter.doFilter(UserTeamsFilter.java:61)
io.hyperfoil.tools.horreum.server.UserTeamsFilter_Subclass.doFilter$$superforward1(UserTeamsFilter_Subclass.zig:118)
io.hyperfoil.tools.horreum.server.UserTeamsFilter_Subclass$$function$$1.apply(UserTeamsFilter_Subclass$$function$$1.zig:47)
io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:132)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:103)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:38)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:57)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:32)
io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:340)
io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
io.hyperfoil.tools.horreum.server.UserTeamsFilter_Subclass.doFilter(UserTeamsFilter_Subclass.zig:195)
javax.servlet.http.HttpFilter.doFilter(HttpFilter.java:97)
io.hyperfoil.tools.horreum.server.UserTeamsFilter_ClientProxy.doFilter(UserTeamsFilter_ClientProxy.zig:284)
io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
io.hyperfoil.tools.horreum.server.GrafanaUserFilter.doFilter(GrafanaUserFilter.java:64)
javax.servlet.http.HttpFilter.doFilter(HttpFilter.java:97)
io.hyperfoil.tools.horreum.server.GrafanaUserFilter_ClientProxy.doFilter(GrafanaUserFilter_ClientProxy.zig:284)
io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:63)
io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:67)
io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:133)
io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:65)
io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:247)
io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:56)
io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:111)
io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:108)
io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
io.quarkus.undertow.runtime.UndertowDeploymentRecorder$9$1.call(UndertowDeploymentRecorder.java:589)
io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:227)
io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:152)
io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:119)
io.undertow.server.Connectors.executeRootHandler(Connectors.java:290)
io.undertow.server.DefaultExchangeHandler.handle(DefaultExchangeHandler.java:18)
io.quarkus.undertow.runtime.UndertowDeploymentRecorder$5$1.run(UndertowDeploymentRecorder.java:415)
java.base@11.0.5/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
java.base@11.0.5/java.util.concurrent.FutureTask.run(FutureTask.java:264)
io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:503)
org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
java.base@11.0.5/java.lang.Thread.run(Thread.java:834)

2021-08-11 11:53:08,945 WARN  [com.arj.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012108: CheckedAction::check - atomic action 0:ffff0a580045:96b7:6111207e:285b aborting with 1 threads active!
2021-08-11 11:53:08,948 WARN  [io.agr.pool] (Transaction Reaper Worker 0) Datasource '<default>': JDBC resources leaked: 0 ResultSet(s) and 1 Statement(s)
2021-08-11 11:53:08,948 WARN  [com.arj.ats.jta] (Transaction Reaper Worker 0) ARJUNA016045: attempted rollback of < formatId=131077, gtrid_length=35, bqual_length=36, tx_uid=0:ffff0a580045:96b7:6111207e:285b, node_name=quarkus, branch_uid=0:ffff0a580045:96b7:6111207e:285e, subordinatenodename=null, eis_name=0 > (io.agroal.narayana.LocalXAResource@7a92143b) failed with exception code XAException.XAER_RMERR: javax.transaction.xa.XAException: Error trying to transactionRollback local transaction: Enlisted connection used without active transaction
	at io.agroal.narayana.LocalXAResource.xaException(LocalXAResource.java:140)
	at io.agroal.narayana.LocalXAResource.xaException(LocalXAResource.java:134)
	at io.agroal.narayana.LocalXAResource.rollback(LocalXAResource.java:88)
	at com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord.topLevelAbort(XAResourceRecord.java:362)
	at com.arjuna.ats.arjuna.coordinator.BasicAction.doAbort(BasicAction.java:3037)
	at com.arjuna.ats.arjuna.coordinator.BasicAction.doAbort(BasicAction.java:3016)
	at com.arjuna.ats.arjuna.coordinator.BasicAction.Abort(BasicAction.java:1679)
	at com.arjuna.ats.arjuna.coordinator.TwoPhaseCoordinator.cancel(TwoPhaseCoordinator.java:124)
	at com.arjuna.ats.arjuna.AtomicAction.cancel(AtomicAction.java:215)
	at com.arjuna.ats.arjuna.coordinator.TransactionReaper.doWork(TransactionReaper.java:431)
	at com.arjuna.ats.internal.arjuna.coordinator.ReaperWorkerThread.run(ReaperWorkerThread.java:78)
Caused by: java.sql.SQLException: Enlisted connection used without active transaction
	at io.agroal.pool.ConnectionHandler.verifyEnlistment(ConnectionHandler.java:362)
	at io.agroal.pool.ConnectionHandler.transactionRollback(ConnectionHandler.java:337)
	at io.agroal.narayana.LocalXAResource.rollback(LocalXAResource.java:85)
	... 8 more

2021-08-11 11:53:08,965 WARN  [org.hib.eng.jdb.spi.SqlExceptionHelper] (executor-thread-140) SQL Error: 0, SQLState: 57014
2021-08-11 11:53:08,966 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (executor-thread-140) ERROR: canceling statement due to user request
  Where: SQL statement "DELETE FROM view_data WHERE runid = OLD.runid"
PL/pgSQL function vd_before_delete_run_func() line 3 at SQL statement
SQL statement "DELETE FROM run_schemas WHERE schemaid = OLD.id"
PL/pgSQL function before_schema_update_func() line 3 at SQL statement
2021-08-11 11:53:08,966 WARN  [org.hib.res.tra.bac.jta.int.syn.SynchronizationCallbackCoordinatorTrackingImpl] (Transaction Reaper Worker 0) HHH000451: Transaction afterCompletion called by a background thread; delaying afterCompletion processing until the original thread can handle it. [status=4]
2021-08-11 11:53:08,966 WARN  [com.arj.ats.arjuna] (Transaction Reaper Worker 0) ARJUNA012121: TransactionReaper::doCancellations worker Thread[Transaction Reaper Worker 0,5,main] successfully canceled TX 0:ffff0a580045:96b7:6111207e:285b
2021-08-11 11:53:08,966 WARN  [com.arj.ats.arjuna] (executor-thread-140) ARJUNA012077: Abort called on already aborted atomic action 0:ffff0a580045:96b7:6111207e:285b
2021-08-11 11:53:08,966 ERROR [io.und.req.io] (executor-thread-140) Exception handling request 23782cb5-1e21-4ae1-9066-46b1be68347a-13 to /api/schema: io.quarkus.arc.ArcUndeclaredThrowableException: Error invoking subclass method
	at io.hyperfoil.tools.horreum.server.UserTeamsFilter_Subclass.doFilter(UserTeamsFilter_Subclass.zig:234)
	at javax.servlet.http.HttpFilter.doFilter(HttpFilter.java:97)
	at io.hyperfoil.tools.horreum.server.UserTeamsFilter_ClientProxy.doFilter(UserTeamsFilter_ClientProxy.zig:284)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.hyperfoil.tools.horreum.server.GrafanaUserFilter.doFilter(GrafanaUserFilter.java:64)
	at javax.servlet.http.HttpFilter.doFilter(HttpFilter.java:97)
	at io.hyperfoil.tools.horreum.server.GrafanaUserFilter_ClientProxy.doFilter(GrafanaUserFilter_ClientProxy.zig:284)
	at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:63)
	at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:67)
	at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:133)
	at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:65)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:247)
	at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:56)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:111)
	at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:108)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$9$1.call(UndertowDeploymentRecorder.java:589)
	at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:227)
	at io.undertow.servlet.handlers.ServletInitialHandler.handleRequest(ServletInitialHandler.java:152)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$1.handleRequest(UndertowDeploymentRecorder.java:119)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:290)
	at io.undertow.server.DefaultExchangeHandler.handle(DefaultExchangeHandler.java:18)
	at io.quarkus.undertow.runtime.UndertowDeploymentRecorder$5$1.run(UndertowDeploymentRecorder.java:415)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:503)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.transaction.RollbackException: ARJUNA016102: The transaction is not active! Uid is 0:ffff0a580045:96b7:6111207e:285b
	at com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.commitAndDisassociate(TransactionImple.java:1285)
	at com.arjuna.ats.internal.jta.transaction.arjunacore.BaseTransaction.commit(BaseTransaction.java:128)
	at io.quarkus.narayana.jta.runtime.CDIDelegatingTransactionManager.commit(CDIDelegatingTransactionManager.java:102)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.endTransaction(TransactionalInterceptorBase.java:365)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:169)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:103)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:38)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:57)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:32)
	at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:340)
	at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
	at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
	at io.hyperfoil.tools.horreum.server.UserTeamsFilter_Subclass.doFilter(UserTeamsFilter_Subclass.zig:195)
	... 47 more


```

**Comment by rvansa** [comment]
It seems that the transaction is timing out because as you update the schema view results for all stored runs is being synchronously updated through triggers. I'll see what I can do about it.

**Comment by rvansa** [comment]
Fixed in 86ac616a23d0e23a97d78eb7dd858fea0f75089d

The problem was missing index in one of triggers running in cascade when you modify schema URI (when you save the schema Hibernate updates all columns, there's no dirty checking on properties). The query that took 160 seconds now runs < 1 second, the update takes another 5-6 seconds which is bearable.

We will eventually need a plan when there's more than few thousands of runs for given schema, e.g. tagging then outdated run_schemas and updating these in a background process spanning multiple transactions.
